src/main/java/org/takes/facets/auth/PsAll.java

```diff
@@ -100,6 +82,28 @@ public final Response exit(final Response response, final Identity identity)
     }
 
     /**
+     * Validate index.
+     * @param idx Index of a Pass whose Identity to return on successful
+     *  {@link PsAll#enter(Request)}
+     * @return Validated index
+     */
+    private int validated(final int idx) {
+        if (idx < 0) {
+            throw new IllegalArgumentException("Index must be >= 0");
```

Comment:
@xupyprmv would be nice to show the value provided, see http://www.yegor256.com/2015/12/01/rethrow-exceptions.html